pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -376,7 +390,8 @@ func (n *NetworkPolicyController) filterAddressGroupsForNamespace(namespace *v1.
 	addressGroups, _ := n.addressGroupStore.GetByIndex(cache.NamespaceIndex, "")
 	for _, group := range addressGroups {
 		addrGroup := group.(*antreatypes.AddressGroup)
-		if addrGroup.Selector.NamespaceSelector.Matches(labels.Set(namespace.Labels)) {
+		// Cluster scoped AddressGroup will not have NamespaceSelector.
+		if addrGroup.Selector.NamespaceSelector != nil && addrGroup.Selector.NamespaceSelector.Matches(labels.Set(namespace.Labels)) {
```

Comment:
Then what does `NamespaceSelector == nil` mean for `Cluster scoped AddressGroup`? matching all namespaces or none?

Reply:
all namespaces. However I do not think filtering out the cluster addrGroups w/o namespaceSelectors is necessary here. If the addrGroup have pod/ee selector, their relative add/update/delete handler to enqueue the addrGroup.

Reply:
I can understand after reading your reply, maybe good to add it to code comment too.
Also the current comments are little confusing, in L389 it says only **cluster scoped groups** can select this namespace because at that time either namespace or namespaceSelector must be set. The new comment should say "Cluster scoped AddressGroups that are created by ClusterNetworkPolicy may have no NamespaceSelector", right?

Reply:
Thanks @tnqn . I added this comment, but I should have been more clear. Will follow up on this.

Reply:
Agreed that the comments need a bit update. Will address this in the CNP e2e test PR.